coffee_core/src/nurse/chain.rs

```diff
@@ -48,16 +50,18 @@ pub struct RecoveryChainOfResponsibility {
 impl RecoveryChainOfResponsibility {
     pub async fn new() -> Result<Self, CoffeeError> {
         Ok(Self {
-            handlers: Vec::new(),
+            handlers: vec![Arc::new(GitRepositoryLocallyAbsentStrategy)],
         })
     }
```

Comment:
The chanin usually takes the struct where you are making the operation on. 

If the chain of responsibility make an operation on the database only, you pass the database inside it as a pointer (Arc<..>).

If the chain of responsibility makes operation on a more general object like CoffeeManager, you pass down the coffee manager always in an `Arc` 

Arc without a Mutex inside is like a `const *void` pointer in C, A pointer that can not be modified so, in this case, the strategy should not modify the coffee manager, but just use it as an access point to information.

Reply:
The `handlers` should be a vector of the strategies to deal with corrupt states. right?

I guess the question is what to pass to `self.recovery_strategies.scan().await`  in `coffee.rs` file.
Initially, it appears that we may require mutable access to the CoffeeManager since certain scenarios may necessitate mutations.

For example, A repository in `CoffeeManager` is missing locally but it can't be restored using `git` for a reason or another. In this case, we remove it from the configuration.

Reply:
I see, so I will let you play with it. I will let you play with it

If you need to pass a mutable state I sure you need a `Arc<Mutex<CoffeeManager>>`